app/uk/gov/hmrc/customs/inventorylinking/export/controllers/HeaderValidator.scala

```diff
@@ -80,5 +87,50 @@ class HeaderValidator @Inject()(logger: ExportsLogger) {
     }
   }
 
-}
+  def eitherBadgeIdentifier[A](allowNone: Boolean)(implicit vhr: HasRequest[A] with HasConversationId): Either[ErrorResponse, Option[BadgeIdentifier]] = {
+    val maybeBadgeId: Option[String] = vhr.request.headers.toSimpleMap.get(XBadgeIdentifierHeaderName)
 
+    if (allowNone && maybeBadgeId.isEmpty) {
+      logger.info(s"$XBadgeIdentifierHeaderName header empty and allowed")
+      Right(None)
+    } else {
+      maybeBadgeId.filter(xBadgeIdentifierRegex.findFirstIn(_).nonEmpty).map(b => {
+        logger.info(s"$XBadgeIdentifierHeaderName header passed validation: $b")
+        Some(BadgeIdentifier(b))
+      }
+      ).toRight[ErrorResponse] {
```

Comment:
I think 'map{ b => ...}' is clearer than 'map(b => {...})' save a set of brackets too.

Reply:
Done.